### PR TITLE
Optimize Transpose->Slice regardless of Slice's axes

### PR DESCRIPTION
**Pull request by mateusztabaka**
Signed-off-by: Mateusz Tabaka <mateusz.tabaka@intel.com>

### CODE REVIEW

**Comment by TomWildenhain-Microsoft** [comment]
Feel free to request a review from me when you are ready to have this merged.

**Comment by mateusztabaka** [comment]
I'm ready, so I'd appreciate your comments on this (seems like I can't add reviewers from github GUI, so I guess I can mark my readiness in the comments section).

File: tests/test_optimizers.py [hunk 1]
```diff
@@ -309,33 +311,84 @@ def test_transpose_dequantize_with_axis(self, shape, perm_input, perm_output):
...
+            model_proto = self.make_model(graph, producer_name="onnx-tests")
+            self.run_transpose_compare(["Z1"], {"X": np.random.randn(*input_shape).astype(np.float32)},
+                                       model_proto, remaining_transpose_num=0)
+
+    @parameterized.expand([
+        ([2, 3, 4, 5], [1, 2, 1, 2], [0, 2, 3, 1], [0, 3, 1, 2]),
+        ([2, 3, 4, 5, 6], [1, 2, 1, 2, 1], [0, 2, 3, 4, 1], [0, 4, 1, 2, 3]),
+    ])
+    @check_opset_min_version(10, "Slice in opset 10 can accept dynamic 'start' and 'ends'")
+    def test_transpose_slice_10(self, input_shape, slice_size, perm_input, perm_output):
```
Comment (TomWildenhain-Microsoft) [inline tests/test_optimizers.py:358]:
```suggestion
    def test_transpose_slice_int32(self, input_shape, slice_size, perm_input, perm_output):
```
Nice catch on the type issue.  I know the other tests just have a number but it is generally nice to have a better test description.

File: tests/test_optimizers.py [hunk 2]
```diff
@@ -309,33 +311,84 @@ def test_transpose_dequantize_with_axis(self, shape, perm_input, perm_output):
...
-        axes = np.array(list(range(len(input_shape))), dtype=np.int64)
-        node1 = helper.make_node("Transpose", ["X"], ["Y"], perm=perm_input, name="trans_1")
-        node2 = helper.make_node("Slice", ["Y", "starts", "ends", "axes"], ["Z"], name="relu")
-        node3 = helper.make_node("Transpose", ["Z"], ["Z1"], perm=perm_output, name="trans_2")
+    @check_opset_max_version(9, "Slice in opset 9 and takes 'axes, 'start' and 'ends' as attributes")
+    def test_transpose_slice(self, input_shape, slice_size, perm_input, perm_output):
+        axes_combinations = []
+        axes = list(range(len(input_shape)))
+        for i in range(1, len(input_shape) + 1):
+            axes_combinations.extend(list(itertools.combinations(axes, i)))
```
Comment (TomWildenhain-Microsoft) [inline tests/test_optimizers.py:322]: Do we really need to test every single combination?  That's like 32 runs per test.  Glad to know they all pass, but I think now we can safely just add an arg for parameterized "axes" and fill in a few tests, get rid of the loop here.

**Review by TomWildenhain-Microsoft — CHANGES_REQUESTED** [review]
Thanks for the contribution!  Some small changes with the tests.  Now that it's reviewed, please submit updates to a new commit instead of amending this one.

File: tests/test_optimizers.py (hunk 1 above)
Comment (mateusztabaka) [reply tests/test_optimizers.py:358]: this test was primarly designed to test Slice from opset 10 (hence the name similarly to test_const_fold_unsqueeze_with_const_13 test case we have). After pretrained_model_test blew up, I decided to change the axes type to int32 in this test case.
Maybe this case should be named like test_transpose_slice_opset_10 - let me know what you think.

File: tests/test_optimizers.py (hunk 2 above)
Comment (mateusztabaka) [reply tests/test_optimizers.py:322]: sure, I will add parametrized axes.

File: tests/test_optimizers.py (hunk 1 above)
Comment (TomWildenhain-Microsoft) [reply tests/test_optimizers.py:358]: Ah, my mistake.  Then yes, test_transpose_slice_opset_10 is great.

**Review by TomWildenhain-Microsoft — APPROVED** [review]

**Pull request merged**